libglusterfs/src/mem-pool.c

```diff
@@ -578,58 +578,44 @@ mem_pools_postfini(void)
      *       so the memory will be released anyway by the system. */
 }
 
-/* Call mem_pools_init() once threading has been configured completely. This
- * prevent the pool_sweeper thread from getting killed once the main() thread
- * exits during deamonizing. */
+static void
+mem_pools_init_once(void)
+{
+    GF_ATOMIC_INIT(sweeper_quit, 0);
+    if (gf_thread_create(&sweeper_tid, NULL, pool_sweeper, NULL, "memsweep"))
+        GF_ABORT("can't create memsweep thread");
+}
+
 void
 mem_pools_init(void)
 {
-    pthread_mutex_lock(&init_mutex);
-    if ((init_count++) == 0) {
-        (void)gf_thread_create(&sweeper_tid, NULL, pool_sweeper, NULL,
-                               "memsweep");
+    static pthread_once_t init_once = PTHREAD_ONCE_INIT;
+    pthread_once(&init_once, mem_pools_init_once);
```

Comment:
Can this function be called more than once ?

Reply:
What about changelog? This is the only xlator which explicitly calls `mem_pools_init()` and `mem_pools_fini()` (no ideas why).